hack/update-codegen.sh

```diff
@@ -57,12 +57,17 @@ source ${REPO_ROOT_DIR}/hack/test-env.sh
 template=$(cat ${REPO_ROOT_DIR}/docs/.test-version.template)
 eval "echo \"${template}\"" > ${REPO_ROOT_DIR}/docs/test-version.md
 
-group "Update gateway API CRDs"
+group "Update Gateway API CRDs"
 
 if command -v kubectl &> /dev/null
 then
-  echo "# Generated with \"kubectl kustomize https://github.com/kubernetes-sigs/gateway-api/config/crd?ref=${GATEWAY_API_VERSION}\"" > ${REPO_ROOT_DIR}/config/100-gateway-api.yaml
-  kubectl kustomize "https://github.com/kubernetes-sigs/gateway-api/config/crd?ref=${GATEWAY_API_VERSION}" >> ${REPO_ROOT_DIR}/config/100-gateway-api.yaml
+ echo "# Generated with \"kubectl kustomize github.com/kubernetes-sigs/gateway-api/config/crd/experimental?ref=${GATEWAY_API_VERSION}" > "${REPO_ROOT_DIR}/config/100-gateway-api.yaml"
+	kubectl kustomize "github.com/kubernetes-sigs/gateway-api/config/crd/experimental?ref=${GATEWAY_API_VERSION}" >> "${REPO_ROOT_DIR}/config/100-gateway-api.yaml"
```

Comment:
It looks like the `100-gateway-api.yaml` is usually auto-created. Maybe we should `.gitignore` it?

Reply:
I guess i want to roll back and understand what this file is intended for. Would like to get a read from @nak3, since he paved the way here and might have some thoughts.

To recap:

What is in `main` now:
- this file is generated by codegen but not being used anywhere as far as I can tell. I assumed it was serving as a local example of what this component is working with for any given release of the Gateway API that we use.

What changes in this PR:
- codegen is updated to generate a new set of CRDs based on the upcoming `v0.5.0` version (changes a bit).
- since this file was there anyway, I changed whatever was pointing at the the upstream (DEVELOPMENT.md and the Istio conformance test) to point at the local file instead. Might be [my beginner roots speaking](https://github.com/knative-sandbox/net-gateway-api/pull/311?w=1#discussion_r906725330) from the era when network was sparse and expensive to nonexisting (and the dinosaurs roamed the earth!) but I still think it's more practical for development.

If pointing to upstream to get a copy of these resources has an advantage over pointing at a local copy (which can be refreshed, to be sure) that I'm not seeing, I can change those references to point upstream. 

If this file ends up having no solid purpose for our use at the moment, I would say maybe not even keep maintaining this code generation and the overhead in time that it adds to running that script?

Reply:
I'm concerned that having this file checked in to `config` (vs `third_party` or somewhere else) will cause it to be collected and bundled into the release.

I can understand wanting to vendor this somewhere rather than fetch it from the Internet; my primary concern is putting it next to the configs we release with `ko resolve -f $DIR`.

Reply:
Sorry yes we should not add the gateway-api CRD into `config` directory but should add it into `third_party` as Evan mentioned.

The reason why I added it into `config` (actually release `net-gateway-api.yaml`) was that we must install the GatewayAPI CRD before the net-gateway-api controller and Istio/Contour. So, I struggled to modify the `ytt` & `kapp` in serving repo and in the end, I added it into the head of [`net-gateway-api.yaml`](https://github.com/knative/serving/blob/main/third_party/gateway-api-latest/net-gateway-api.yaml) but I am sure that is not ideal.

Probably you have knowledge for the two vmware CLI tools, so I would like you to move the file into `third_party` and modify the test script in serving repo.